app/backend/src/couchers/servicers/groups.py

```diff
@@ -199,8 +211,8 @@ def JoinGroup(self, request, context):
             if not cluster:
                 context.abort(grpc.StatusCode.NOT_FOUND, errors.GROUP_NOT_FOUND)
 
-            current_membership = cluster.members.filter(User.id == context.user_id).one_or_none()
-            if current_membership:
+            user_in_group = cluster.members.filter(User.id == context.user_id).one_or_none()
+            if user_in_group:
```

Comment:
Please don't change this if there's no good reason.

Reply:
I changed it as part of a bigger change that got rolled back. I think it's clearer, but doesn't really matter

Reply:
There's a few places where this kind of stuff is done (`if [not] (...).one_or_none`), maybe we should write a quick wrapper helper or just start standardizing. This is also wasteful currently: the whole user gets shipped to the backend. Something like `if (...).count() [!/=]= 0` or something...

Anyway, leave this if you don't feel like cleaning up, it's fine either way (user_in_group or current_membership)

Reply:
I'm not understanding your first comment

Happy to change back while I clean up the rest, like I said, doesn't really matter

Reply:
No need to change it back, it's good as is.

I mean that there are several places where all we do is check whether some object exists (in here, whether there is a `User` that matches that filter in `cluster.members`). Doing `one_or_none()` does a `SELECT * FROM ... WHERE ...` that fetches the whole `User` object from the database — just so we can check whether that object is `None` or not. That's pretty wasteful, instead we'd want to do something like `SELECT count(*) FROM ... WHERE ...`. We do this in quite a few different places, and should probably refactor the full `SELECT *` out and replace with `SELECT count()*`. (That's outside the scope of this PR though and should be done separately.)

Reply:
Is count quicker? If so, then ya, we do it a lot. Make an issue for it. Probably good first issue.

Reply:
Yeah, `count(*)` just returns ~1 bytes + overhead to the backend from the database (and can probably be done using only an index, without having to visit the table). `*` returns the whole user object to the backend and requires definitely a table access. (Also, `User` models has some `column_property`s that also need to be computed whenever we grab the whole thing.)

This requires a tad bit of investigation though, to make sure SQLAlchemy doesn't do something dumb (like `SELECT count(*) FROM (SELECT *) AS t ...` which would negate a bunch of the point. Will open an issue.